### PR TITLE
add support for outputting msg.Fields from s3_parquet.lua

### PR DESCRIPTION
**Pull request by relud**
intended for use in conjunction with #79

### CODE REVIEW

File: parquet/sandboxes/heka/output/s3_parquet.lua [hunk 1]
```diff
@@ -126,7 +130,8 @@ if type(json_objects) == "table" then
     require "cjson"
     json_objects_len = #json_objects
 end
-if not json_objects and metadata_group then error("metadata_group cannot be configured without json_objects") end
+local heka_msg_use_fields   = read_config("heka_message_use_fields")
+if not json_objects and not heka_msg_use_fields and metadata_group then error("metadata_group cannot be configured without json_objects") end
```
Comment (trink) [inline parquet/sandboxes/heka/output/s3_parquet.lua]: This reads a bit weird.. you will need different checks as this one allows both json_objects and heka_msg_use_fields
and a metadata/heka_msg_use_fields combo

File: parquet/sandboxes/heka/output/s3_parquet.lua [hunk 2]
```diff
@@ -283,10 +288,38 @@ local function load_json_objects()
     return root, nil
 end
 
+local function load_msg_fields()
+    local raw = decode_message(read_message("raw")).Fields
```
Comment (trink) [inline parquet/sandboxes/heka/output/s3_parquet.lua]: fields would be a better name

Comment (relud) [inline parquet/sandboxes/heka/output/s3_parquet.lua]: good point, will fix

File: parquet/sandboxes/heka/output/s3_parquet.lua (hunk 1 above)
Comment (relud) [inline parquet/sandboxes/heka/output/s3_parquet.lua]: i'll make them separate checks

**Comment by trink** [comment]
I am still unclear why you don't use the Heka message schema is it simply to avoid a column name of Fields.Foo and just have a column called Foo?  If so I really don't think it is work taking a performance hit of 5-10x

**Comment by relud** [comment]
It's about preserving message metadata. #79 puts everything in `Fields` to preserve metadata, and this pulls everything out of `Fields`, to get the same behavior

**Comment by trink** [comment]
What does the parquet schema look like are you putting it all in a map?

**Comment by relud** [comment]
right now schemas we are using look like this: https://github.com/mozilla/normandy/blob/master/recipe-server/parquet_schemas/normandy.app.docker.app/request.summary.cfg#L12-L39

**Comment by relud** [comment]
but that requires code not in this PR, like this:
<pre>
    local raw_fields = decode_message(read_message("raw")).Fields
    -- if the message didn't have Fields then we can't we can't encode the message, skip it
    if not raw_fields then return nil end

    -- convert from raw message table fields to a hash http://mozilla-services.github.io/lua_sandbox/heka/message.html#array-based-message-fields
    msg = {}
    for _, field in ipairs(raw_fields) do
        if #field.value == 1 then
            msg[field.name] = field.value[1]
        else
            msg[field.name] = field.value
        end
    end

    -- scrub agent
    for field, prefix in pairs(cfg.user_agent_transforms) do
        if msg[field] then
            msg[prefix.."browser"],
            msg[prefix.."version"],
            msg[prefix.."os"] = clf.normalize_user_agent(msg[field])
            if not ((cfg.user_agent_conditional and not msg[prefix.."browser"]) or cfg.user_agent_keep) then
                msg[field] = nil
            end
        end
    end

    orig = msg
    msg = {Fields={}}
    for key, value in pairs(orig) do
        if key:sub(1,7) == "Fields." then
            msg[key] = nil
            key = key:lower():gsub("[.-]","_")
            msg[key] = value
            if type(value) == 'table' then
                msg.Fields[key:sub(8, #key)] = concat(value, ",")
            else
                msg.Fields[key:sub(8, #key)] = tostring(value)
            end
        else
            key = key:gsub("[.-]","_")
            msg[key] = value
        end
    end
</pre>

**Comment by relud** [comment]
This isn't a generic use case, closing wontfix

**Pull request closed without merging**